src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -51,7 +51,7 @@ public abstract class NLPProcessor extends AbstractProcessor {
 
     private final Environment environment;
 
-    public NLPProcessor(
+    public InferenceProcessor(
```

Comment:
Better naming would be TextInferenceProcessor as we dealing with Texts in these processors.

Reply:
There are several considerations:
1. This is an abstract parent class and it's better to be generic.
2. Most of the code in this class are to create corresponding data structure to ml-commons which is generic logic.
3. ml-commons is supporting more inference inputs like image(base64), when neural search going to support this, we can create a new class to extend this since this is generic.